src/gl.rs

```diff
@@ -63,6 +63,30 @@ fn calculate_length(width: GLsizei, height: GLsizei, format: GLenum, pixel_type:
     return (width * height * colors * depth) as usize;
 }
 
+// https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.10
+fn get_uniform_vector_length(location: &GLint) -> usize {
+    match *location as u32 {
+        ffi::BOOL |
+        ffi::INT |
+        ffi::FLOAT |
+        ffi::SAMPLER_2D |
+        ffi::SAMPLER_CUBE => 1,
+        ffi::INT_VEC2 |
+        ffi::BOOL_VEC2 |
+        ffi::FLOAT_VEC2 => 2,
+        ffi::INT_VEC3 |
+        ffi::BOOL_VEC3 |
+        ffi::FLOAT_VEC3 => 3,
+        ffi::INT_VEC4 |
+        ffi::BOOL_VEC4 |
+        ffi::FLOAT_VEC4 |
+        ffi::FLOAT_MAT2 => 4,
+        ffi::FLOAT_MAT3 => 9,
+        ffi::FLOAT_MAT4 => 16,
+        _ => panic!("This shouldn't happen"),
+    }
```

Comment:
Would it be worth separating this into two functions? For example, since we never pass `location` which is `GL::FLOAT` to `glGetUniformiv`, it will never match, and vice versa. Separating it into `get_uniform_iv_vector_length` and `get_uniform_fv_vector_length` could be a bit more efficient. What do you think?